### PR TITLE
Serializers

### LINKED ISSUE #16: Ability to set serializer for server and client
Serializer should implement `dumps` and `loads` functions. Default serializer is marshal.

### PR DESCRIPTION
**Pull request by Gelbpunkt**
fixes #16 

This adds new kwargs to client and server: `loads & dumps`. It defaults to marshal's. The typehints are that way because e.g. marshal only accepts bytes, so we have to use a Union for functions that work with bytes and those with strings.

I tested this with marshal and json.

Review would be amazing!

### CODE REVIEW

File: iomirea_rpc/client.py [hunk 1]
```diff
@@ -33,6 +32,8 @@
 
 log = logging.getLogger(__name__)
 
+_SerializerType = Any  # see server.py
```
Comment (Fogapod) [inline iomirea_rpc/client.py]: Can't you create abstract class with `dumps` and `loads` method instead?

File: iomirea_rpc/client.py [hunk 2]
```diff
@@ -148,6 +150,16 @@ def __init__(
         else:
             self._loop = loop
 
+        if serializer is None:
+            import marshal
+
+            self._serializer = marshal
+        else:
+            if not hasattr(serializer, "loads") or not hasattr(serializer, "dumps"):
+                raise ValueError("Serializer does not implement loads and dumps.")
```
Comment (Fogapod) [inline iomirea_rpc/client.py]: Would be a good idea to also check if these are callable

File: iomirea_rpc/server.py [hunk 3]
```diff
@@ -116,14 +130,14 @@ def remove_command(self, index: int) -> _CommandType:
         await self._handler(channels[0])
 
     def run(self, *args: Any, **kwargs: Any) -> None:
-        """A blocking way to start server. Takes same arguments as Server.start."""
+        """A blocking way to start server. Takes same arguments as Client.start."""
```
Comment (Fogapod) [inline iomirea_rpc/server.py]: I think it is `Server`

**Comment by Fogapod** [comment]
Looks good

File: iomirea_rpc/server.py (hunk 3 above)
Comment (Gelbpunkt) [reply iomirea_rpc/server.py]: Woops seems I messed up there

Comment (Gelbpunkt) [reply iomirea_rpc/server.py]: Actually, no. It's in the server.py file

**Pull request merged**